File: src/main/java/edu/kit/ifv/mobitopp/actitopp/CSVWRDDistributionLoader.java
package edu.kit.ifv.mobitopp.actitopp;


import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
/**
 * 
 * @author Tim Hilgert
 *
 */
public class CSVWRDDistributionLoader
{

  public WRDModelDistributionInformation loadDistributionInformation(InputStream input)
  {
  	WRDModelDistributionInformation minfo = new WRDModelDistributionInformation();
  	   
    try (BufferedReader inRead = new BufferedReader(new InputStreamReader(input)))
    {
      boolean header = true;
      String line = null;

      while ((line = inRead.readLine()) != null)
      {
        if (header)
        {
          line = inRead.readLine();
          header = false;
        }
        String[] splitted = line.split(";");
        
        int slot = Integer.parseInt(splitted[0]);
        int amount = Integer.parseInt(splitted[1]);
        
        minfo.addDistributionElement(slot, amount);
        
      }      
    } 
    catch (IOException e) 
    {
			e.printStackTrace();
		}
    return minfo;
  }
}

File: src/main/java/edu/kit/ifv/mobitopp/actitopp/CSVWRDDistributionLoader.kt
package edu.kit.ifv.mobitopp.actitopp

import java.nio.file.Path
import kotlin.io.path.Path
import kotlin.io.path.exists
import kotlin.io.path.useLines


fun loadDistributionInformationFromFile(string: String) =loadDistributionInformationFromFile(Path(string))
fun loadDistributionInformationFromFile(path: Path): WRDModelDistributionInformation {
    require(path.exists()) {
        "The path $path does not exist."
    }
    return path.useLines { lines ->
        val map = lines.drop(1).associate { line ->
            val split = line.split(";")
            val slot = split[0].toInt()
            val amount = split[1].toInt()
            slot to amount
        }
        WRDModelDistributionInformation(map)
    }


}